tests/test_plugin_loader_runner.py

```
@@ -0,0 +1,87 @@
+            mock_subprocess.side_effect = [
+                subprocess.CalledProcessError(1, ["git", "status"]),
+                subprocess.CompletedProcess(args=["git", "status"], returncode=0),
+            ]
+            result = _run(["git", "status"], retry_attempts=2, retry_delay=0)
+            self.assertIsInstance(result, subprocess.CompletedProcess)
+            self.assertEqual(mock_subprocess.call_count, 2)
+
+    def test_run_raises_after_max_attempts(self):
+        with patch("subprocess.run") as mock_subprocess:
+            mock_subprocess.side_effect = subprocess.CalledProcessError(1, ["git"])
+            with self.assertRaises(subprocess.CalledProcessError):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)